checked/debug.go

```diff
@@ -114,24 +128,31 @@ type debugger struct {
 	entries [][]*debuggerEntry
 }
 
-func (d *debugger) append(event debuggerEvent, ref int, stack []byte) {
+func (d *debugger) append(event debuggerEvent, ref int, pc []uintptr) {
 	if len(d.entries) == 0 {
 		d.entries = make([][]*debuggerEntry, 1, tracebackCycles)
 	}
 	idx := len(d.entries) - 1
-	d.entries[idx] = append(d.entries[idx], &debuggerEntry{
-		event: event,
-		ref:   ref,
-		stack: stack,
-		t:     time.Now(),
-	})
+	entry := tracebackEntryPool.Get().(*debuggerEntry)
+	entry.event = event
+	entry.ref = ref
+	entry.pc = pc
+	entry.t = time.Now()
+	d.entries[idx] = append(d.entries[idx], entry)
 	if event == decRefEvent && ref == 0 {
 		if len(d.entries) == tracebackCycles {
 			// Shift all tracebacks back one if at end of traceback cycles
+			slice := d.entries[0]
+			for i, entry := range slice {
+				tracebackCallersPool.Put(entry.pc)
```

Comment:
I wonder what this code would actually do? `entry.pc` is a `[]uintptr`, i.e. it's a struct with a pointer to a vector on the heap. When we put it back into the pool, it gets casted into an `interface{}`, i.e. now it's a pointer to a struct with a pointer to a vector on the heap, and it ends up being stored somewhere in the pool. Then we nil it, effectively setting the pointer in the struct to nil and releasing the vector. Does it mean that inside the pool we now have a struct with a nil pointer, thus not really pooling the actual heap memory but only the slice headers?